docs/reference/completions.md

```diff
@@ -108,3 +111,44 @@ you have to escape it like this: `\\$`
 (double backslashes are needed
 because we are within a JSON string).
 :::
+
+
+## Completions Metadata
+
+``` json
+{
+  "trigger": "func",
+  "contents": "funcbar",
+  "annotation": "function",
+  "kind": "function",
+  "details": "A short description of what this string function does.",
+}
+```
+
+These do not affect the triggers themselves,
+but allow for customization of the appearance of completions
+in the completions list.
+
+**annotation** {#annotation}
+: Displays as right-aligned gray text to the right of the entry in the completions list.
+  Does not affect the trigger itself.
+
+  Annotations can also be defined
+  using a tab character `\t` in [`trigger`](#trigger).
+
+  ::: tip Added in build 4050 {added}
```

Comment:
Is there a way to communicate that the `\t` annotation was available earlier, but the `annotation` key came in 4050?